src/zgamma_producer.cpp

```diff
@@ -56,13 +74,21 @@ void ZGammaVarProducer::WriteZGammaVars(nano_tree &nano, pico_tree &pico, vector
         dijet = j1 + j2;
         pico.out_llphoton_dijet_dphi().push_back(llg.DeltaPhi(dijet));
         pico.out_llphoton_dijet_balance().push_back((dilep+photon+j1+j2).Pt()/(dilep.Pt()+photon.Pt()+j1.Pt()+j2.Pt()));
+        pico.out_llphoton_dijet_dr().push_back(llg.DeltaR(dijet));
         pico.out_photon_jet_mindr().push_back(min(photon.DeltaR(j1), photon.DeltaR(j2)));
+        pico.out_photon_jet_maxdr().push_back(max(photon.DeltaR(j1), photon.DeltaR(j2)));
+        pico.out_photon_jet1_dr().push_back(photon.DeltaR(j1));
+        pico.out_photon_jet2_dr().push_back(photon.DeltaR(j2));
+
         pico.out_photon_zeppenfeld().push_back(abs(photon.Eta() - (j1.Eta() + j2.Eta())/2));
+
         TVector3 g = photon.Vect();
         TVector3 h = llg.Vect();
         TVector3 z = dilep.Vect();
         g.SetZ(0); h.SetZ(0); z.SetZ(0);
-        pico.out_llphoton_pTt2().push_back(h.Cross(z-g).Mag()/h.Mag());
+        pico.out_llphoton_pTt().push_back( fabs(h.Cross(z-g).Pz())/(z-g).Pt());
+        pico.out_llphoton_pTt_an_hig019014().push_back( fabs(h.Cross(z-g).Pz())/h.Mag() );
+        pico.out_llphoton_pTt_oldpicos().push_back(h.Cross(z-g).Mag()/h.Mag());
```

Comment:
I believe the correct way in calculating ptt is,

h.Cross((z-g).Unit()).Mag()

The way that hig19-014 calculated this was "h.Cross(z-g).Mag()/h.Mag()"

Let's not have 3 calculation methods. Let's just have the correct ptt and the ptt calculated by hig19-014.

Reply:
Do we want the hig-019-014 version  of the variable in the AN or the paper.
The paper version is the first of the three there, and the AN version is on line 90.

I dont believe that the h.Cross((z-g).Unit()).Mag() is correct though because h,z,g are 3D vectors where in the paper they only care about the cross product of the pT of the higgs candidate and the Z-g object (see below). The denominator on line 90 should be h.Pt() though.

AN definition: ![image](https://github.com/richstu/nano2pico/assets/103283186/b1e8289e-3703-489b-9868-7ebee512b047)
Paper: ![image](https://github.com/richstu/nano2pico/assets/103283186/f53dfb4e-98f2-40ad-82b2-a8f4618e07d2)

Reply:
Line 88 sets the gamma, higgs and Z to be the pT "vector". So I think "h.Cross((z-g).Unit()).Mag()" is correct.

I don't care that much of what is written in the AN. I think we should have a version that was used in the HIG19014 analysis code, which was the ptt2 variable.